it/proxy_test.py

```diff
@@ -0,0 +1,95 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import collections
+import os
+import shutil
+import tempfile
+
+import pytest
+
+import it
+from esrally.utils import process
+
+HttpProxy = collections.namedtuple("HttpProxy", ["authenticated_url", "anonymous_url"])
+
+
+@pytest.fixture(scope="module")
+def http_proxy():
+    config_dir = os.path.join(os.path.dirname(__file__), "resources", "squid")
+
+    lines = process.run_subprocess_with_output(f"docker run --rm --name squid -d "
+                                               f"-v {config_dir}/squidpasswords:/etc/squid/squidpasswords "
+                                               f"-v {config_dir}/squid.conf:/etc/squid/squid.conf "
+                                               f"-p 3128:3128 datadog/squid")
+    proxy_container_id = lines[0].strip()
+    proxy = HttpProxy(authenticated_url=f"http://testuser:testuser@127.0.0.1:3128",
+                      anonymous_url=f"http://127.0.0.1:3128")
+    yield proxy
+    process.run_subprocess(f"docker stop {proxy_container_id}")
+    process.run_subprocess(f"docker rm {proxy_container_id}")
+
+
+# ensures that a fresh log file is available
+@pytest.fixture(scope="function")
+def fresh_log_file():
+    cfg = it.ConfigFile(config_name=None)
+    log_file = os.path.join(cfg.rally_home, "logs", "rally.log")
+
+    if os.path.exists(log_file):
```

Comment:
just curious, why would we not just zero out the log contents if we want a new logfile every time, instead of moving its contents. I assume the contents of the "last run" are useless if its a passing test by the time you get to the next test.

Reply:
The idea behind this is that we should keep the full history of the log file. I am not so concerned with CI but rather when you run the tests locally. It might be puzzling (and maybe even a problem) if we start truncating the log file.

Reply:
oh ok this makes sense